src/main/java/mate/academy/service/impl/OrderServiceImpl.java

```diff
@@ -0,0 +1,33 @@
+package mate.academy.service.impl;
+
+import java.time.LocalDateTime;
+import java.util.List;
+import mate.academy.dao.OrderDao;
+import mate.academy.lib.Inject;
+import mate.academy.lib.Service;
+import mate.academy.model.Order;
+import mate.academy.model.ShoppingCart;
+import mate.academy.model.User;
+import mate.academy.service.OrderService;
+import mate.academy.service.ShoppingCartService;
+
+@Service
+public class OrderServiceImpl implements OrderService {
+    @Inject
+    private OrderDao orderDao;
+    @Inject
+    private ShoppingCartService shoppingCartService;
+
+    @Override
+    public Order completeOrder(ShoppingCart shoppingCart) {
+        Order order = new Order(shoppingCart.getTickets(), LocalDateTime.now(),
+                shoppingCart.getUser());
+        shoppingCartService.clearShoppingCart(shoppingCart);
+        return order;
```

Comment:
```suggestion
        Order order = new Order(new ArrayList<>(shoppingCart.getTickets()), LocalDateTime.now(),
                shoppingCart.getUser());
        orderDao.add(order);
        shoppingCartService.clearShoppingCart(shoppingCart);
        return order;
```
there is an issue with setting tickets to order. Hibernate may delete tickets from shoppingCart(), but we should do it manually by ourselves 
P.S. check the common mistakes file